src/Nuvem.PharmacyManagementSystem.Pharmacy.Api/Program.cs

```diff
@@ -1,21 +1,30 @@
 using System.Net.NetworkInformation;
 using System.Reflection;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
 using Nuvem.PharmacyManagementSystem.Pharmacy.Data;
 using Nuvem.PharmacyManagementSystem.Pharmacy.Data.Data;
 using Nuvem.PharmacyManagementSystem.Pharmacy.Services;
 
 var builder = WebApplication.CreateBuilder(args);
 
-// Add services to the container.
-builder.Services.AddControllers();
+// Add services to the container. 
+builder.Services.AddControllers(options  => 
+{
+    options.Filters.Add(new ProducesAttribute("application/json"));
+    options.Filters.Add(new ConsumesAttribute("application/json"));
+});
+
+
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
 
 Assembly apiAssembly = Assembly.GetExecutingAssembly();
 Assembly servicesAssembly = Assembly.Load("Nuvem.PharmacyManagementSystem.Pharmacy.Services");
 Assembly dataAssembly = Assembly.Load("Nuvem.PharmacyManagementSystem.Pharmacy.Data");
 builder.Services.AddAutoMapper(apiAssembly,servicesAssembly);
 
+builder.Services.AddScoped<IPharmacyDbContext, PharmacyDbContext>();
```

Comment:
Nice to start this way - but why used scoped?
What possible hazard? Any better option?

Reply:
Transient.

Reply:
Nope...

Reply:
Singleton is not thread safe and can generate execptions when used concurrently. And a long-lived DbContext will tend to accumulate lots and lots of objects in its Change Tracker, degrading performance and wasting resources.

The lifetime of a DbContext begins when the instance is created and ends when the instance is [disposed]. A DbContext instance is designed to be used for a single [unit-of-work]. This means that the lifetime of a DbContext instance is usually very short.
(https://learn.microsoft.com/en-us/ef/core/dbcontext-configuration/)